src/components/course/routes/ExternalCourseEnrollment.jsx

```diff
@@ -21,10 +26,10 @@ import { features } from '../../../config';
 const ExternalCourseEnrollment = () => {
   const config = getConfig();
   const navigate = useNavigate();
+  const { courseRunKey } = useParams();
   const { pathname } = useLocation();
   const {
     state: {
-      activeCourseRun,
```

Comment:
One of the behaviors for this component below is to redirect the user from `/enroll` to `/enroll/complete` if the user has already redeemed this particular course: https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/2f5b30e5e92446a5e6863789cdea94b90a4a3eff/src/components/course/routes/ExternalCourseEnrollment.jsx#L71-L78

Currently, `hasSuccessfulRedemption` is based on whether _any_ course run has been redeemed. However, now that this route is for a specific `:courseRunKey` per the route parameter, we might need to update `hasSuccessfulRedemption` to be specific to the course run key in the route params.

Otherwise, if a learner has redeemed a different Exec Ed course run when they reach this route, they will be redirected straight to confirmation/complete route as if they've already redeemed the specified course run (which they might not have).

Will the `checkRedemptionEligibility` function that calls the `can-redeem` API and defines `hasSuccessfulRedemption` need to be modified to consider only the specified `courseRunKey` (when provided) instead of _any_ course run?

Example:

```jsx
const hasSuccessfulRedemption = courseRunKey
  ? !!data.find(r => r.contentKey === courseRunKey)?.hasSuccessfulRedemption
  : data.some(r => r.hasSuccessfulRedemption);
```

The challenge here might be how you access the `courseRunKey` within `checkRedemptionEligibility`. Note that while migrating the `/enroll` route into the new routing paradigm, I ended up having to do something similar to prevent the redirect to `/complete` from happening when it shouldn't.

Reply:
[sanity check] Related, when `completeEnrollmentUrl` is generated below, is it the correct path given the route path now includes `:courseRunKey`? It might need to be passed in like the other route params (e.g., `enterpriseSlug`, `courseKey`, etc.).

Reply:
Re: sanity check: I'm pretty sure that generatePath() call is a no-op passthrough for the input string in this case:

```
> const reactRouterDom = require("react-router-dom");
undefined
> reactRouterDom.generatePath('string/without/pattern/tokens', 'foo', 'bar')
'string/without/pattern/tokens'
```

Reply:
Ah, so when React Router was upgraded from v5 to v6, the usage of `pathname` was introduced, which as you're pointing out, doesn't contain any route params in the path where by `generatePath` is essentially a no-op.

Previously, the implementation was:

```jsx
const routeMatch = useRouteMatch();

const completeEnrollmentUrl = generatePath(
  `${routeMatch.path}/complete`,
  { enterpriseSlug: slug, courseType: course.courseType, courseKey: course.key },
);
```

Your recommendation to no longer use `generatePath` makes sense.